src/main/java/mate/academy/rickandmorty/service/impl/CharacterClientImpl.java

```diff
@@ -0,0 +1,52 @@
+package mate.academy.rickandmorty.service.impl;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.AllArgsConstructor;
+import mate.academy.rickandmorty.dto.external.CharacterResponseDataDto;
+import mate.academy.rickandmorty.dto.external.CharacterResponseDto;
+import mate.academy.rickandmorty.service.CharacterClient;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class CharacterClientImpl implements CharacterClient {
+    private static final String BASE_URL = "https://rickandmortyapi.com/api/character";
+    private final ObjectMapper objectMapper;
+    private final HttpClient httpClient = HttpClient.newHttpClient();
```

Comment:
You have Spring IoC to avoid instantiation like this.
```suggestion
    private final HttpClient httpClient = HttpClient.newHttpClient();
```